backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -134,6 +138,60 @@ public void updateUser(String oldUsername, IdentityAttributes userIdentity) {
         user.update();
     }
 
+    public OrganizationRole updateUserRole(String username, OrganizationRole role) {
+        UserList users = _client.listUsers(username, null, null, null, null);
+        if (!users.iterator().hasNext()) {
+            throw new IllegalGraphqlArgumentException("Cannot update role of Okta user with unrecognized username");
+        }
+        User user = users.single();
+
+        Set<String> orgExternalIds = new HashSet<>();
+
+        // Remove user from old groups
+        for (Group g : user.listGroups()) {
+            String groupName = g.getProfile().getName();
+            for (OrganizationRole r : OrganizationRole.values()) {
+                if (g.getType().equals(GroupType.OKTA_GROUP) &&
+                            groupName.startsWith(_rolePrefix)) {
+                    // Remove user from non-USER groups only
+                    if (!groupName.endsWith(generateRoleSuffix(OrganizationRole.USER))) {
+                        g.removeUser(user.getId());
+                    }
+                    // Keep track of organizations user belongs to
+                    orgExternalIds.add(getOrganizationExternalIdFromGroupName(groupName, r));
+                }
+            }
+        }
+
+        // Add user to new groups
+        for (String orgExternalId : orgExternalIds) {
```

Comment:
The handling for multiple orgs is unexpected. 

From my understanding if a users is in multiple groups across orgs then something has gone wrong and we should raise an exception

either way this seems like it works fine for the case of a user belonging to a single org

Reply:
Ben felt we should push to maintain a user's relationships in deactivated or archived organizations, instead of deleting those relationships outright. Personally, I'm inclined to agree with you, but yeah, because this code accommodates either scenario, I haven't pushed much one way or another.

Reply:
The hitch in this though is that Okta doesn't _have_ a concept of deactivated/archived groups, so I don't believe there is a way to tell Okta "only update the user's relationships with groups that are active".

While I'd like the "only call this method from an org admin so that we can pass in the target group to this method", i don't think in practice we can limit update-user-role to org admins only. we as superadmins need to be able to make someone into the first org-admin of their org, or update their role if it wasn't set correctly at the outset.

Reply:
There isn't a way to tell Okta that, but we as the people calling the method should always be able to know which groups are active, because we know which organization we think the person belongs to (at least, we had better!) and we know which organizations are active on our end. It's not a trivial change, but I think it would actually simplify the code a lot if we required the organization ID as an input, so that we're looking for a specific enumerated set of groups rather than any group that might be a role-conveying group. (I could be talked into worrying about the case when a role has been added or removed such that there's a non-enumerable Okta group name, but it would take some talking.)

Reply:
In general I think that we would be better off with a temporary "permission to act as org admin" mechanism, so we can eliminate this class of "but we need to pass in the org ID explicitly" problem, but that's not where we are.

Reply:
i think for the time being we should leave the "inactive organizations" use case off the table, at the very least because the necessary changes for that would merit their own PR. but i'll change this updateUserRole() method to take an org as a parameter.

Reply:
actually, i take it back, because i just realized the change that i just made to pass an org into `OktaRepository::updateUserRole()` also implicitly validates that that org isn't soft-deleted in postgres :)